aait/backend/group-2A/Infrastructure/Service/AuthService.cs

```diff
@@ -0,0 +1,120 @@
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Infrastructure.Service;
+
+public class AuthService : IAuthService {
+    
+     private readonly UserManager<ApplicaionUser> _userManager;
+     private readonly SignInManager<ApplicaionUser> _signInManager;
+     private readonly JwtSetting _jwtSettings; 
+    
+     public AuthService(UserManager<ApplicaionUser> userManager,SignInManager<ApplicaionUser> signInManager, IOptions<JwtSetting> jwtSettings) 
+     { 
+         _userManager = userManager; 
+         _signInManager = signInManager; 
+         _jwtSettings = jwtSettings.Value; 
+          
+     } 
+     
+     public async Task<string?> Login(AuthRequest request, IUserRepository userRepository) 
+     {
+         var user = await _userManager.FindByEmailAsync(request.Email);
+         if (user is null){
+             Console.WriteLine("-------------------User Not Found ");
+             return null;
+             // throw new NotFoundException("Account",req.Email); 
+         }
+    
+         var isCorrect = await _signInManager.PasswordSignInAsync(userName: user.UserName, request.Password,isPersistent:true, lockoutOnFailure:false);
+         if (!isCorrect.Succeeded){
+             Console.WriteLine("-------------------Creditial Not Found ");
+             return null;
+             // throw new Exception($"invalid credentials for user: {request.Email}"); 
+         }
+         
+         Console.WriteLine("-------------UseR Repository started");
+
+         var customeUser = await userRepository.GetUserByEmail(user.Email);
+
+         Console.WriteLine("---------------OKKKKKKKKKKKKK  JWT");
+         JwtSecurityToken token = await GenerateToken(user, customeUser.Id.ToString());
+         var Token = new JwtSecurityTokenHandler().WriteToken(token);
+         Console.WriteLine("---------------OKKKKKKKKKKKKK  JWT");
+         return Token; 
+     
+     } 
+     
+     public async Task<bool?> Register(CreateUserDTO req, IUserRepository userRepository) 
+     {
+         var alreadyExist = await _userManager.FindByEmailAsync(req.Email); 
+         if (alreadyExist is not null){
+             Console.WriteLine("----Email Exist");
+             return null;
+             // throw new Exception($"{req.Email} already exist"); 
+         } 
+    
+         var user = new ApplicaionUser { 
+             Email = req.Email, 
+             UserName = req.UserName, 
+             EmailConfirmed = true
+         };
+         var creatingUser = await _userManager.CreateAsync(user, req.Password);
+         if (!creatingUser.Succeeded){
+             Console.WriteLine("Create Error ---------------");
+             Console.WriteLine(creatingUser.ToString());
+             return null;
+             // throw new Exception($"Failed to register \n"); 
+         } 
+         Console.WriteLine("User Registered");
+         AuthRequest request = new AuthRequest {Email = req.Email, Password = req.Password };
```

Comment:
since you aren't using  "AuthRequest request = new AuthRequest {Email = req.Email, Password = req.Password }; " delete